src/main.rs

```diff
@@ -137,8 +150,11 @@ fn merge_sort<T: PartialOrd + std::marker::Copy + std::fmt::Debug>(v: &[T]) -> V
     }
     let middle = v.len() / 2; //rounds down by default
     let left = merge_sort(&v[0..middle]);
+    println!("Left array: {:?}", left);
```

Comment:
You want to remove these print statements – they generate a _lot_ of noise if you try to sort anything big.